src/App.tsx

```diff
@@ -1,27 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<Good[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleClick = (getGoods: () => Promise<Good[]>) => {
+    setErrorMessage('');
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    getGoods()
+      .then(setVisibleGoods)
+      .catch((error) => setErrorMessage(error.toString()));
+  };
```

Comment:
Alternative, You can create memoize value to prevent make request each time when you click on button, for example:
const allTodos = useMemo(() => getAll(), []) 
const firstFiveTodos= useMemo(() => get5First(), [])